pkg/endpoint/bpf.go

```diff
@@ -136,31 +137,43 @@ func (e *Endpoint) writeHeaderfile(prefix string) error {
 	e.getLogger().WithFields(logrus.Fields{
 		logfields.Path: headerPath,
 	}).Debug("writing header file")
-	f, err := os.Create(headerPath)
+
+	// Write new contents to a temporary file which will be atomically renamed to the
+	// real file at the end of this function. This will make sure we never end up with
+	// corrupted header files on the filesystem.
+	f, err := renameio.TempFile("", headerPath)
```

Comment:
```
// If dir is the empty string, TempDir(filepath.Base(path)) is used. If you are
// going to write a large number of files to the same file system, store the
// result of TempDir(filepath.Base(path)) and pass it instead of the empty
// string.
```

I think this should be `renameio.TempFile(prefix, headerPath)`/ This way we would at least writing it directly into the host FS. By leaving it as `renameio.TempFile("", headerPath)` we might write it into the container's tmp directory where the `mv` might fail when moving the file from the container's FS into the host FS.

Reply:
Nice catch, thanks. Looks like I didn't read the godoc careful enough. Will adjust to `renameio.TempFile(prefix, headerPath)` as suggested.